File: mongo/java/update/mongo_update.java
// --------------------------------------------------------------
//	update/mongo_update.java
//
//					Sep/11/2015
// --------------------------------------------------------------
import	java.util.HashMap;

import com.mongodb.MongoClient;
import com.mongodb.client.MongoDatabase;
import com.mongodb.client.MongoCollection;
import com.mongodb.client.FindIterable;
import com.mongodb.Block;

import com.mongodb.BasicDBObject;

import org.bson.Document;

// --------------------------------------------------------------
public class mongo_update
{
// --------------------------------------------------------------
public static void mongo_update_proc
	(MongoCollection coll,String key_in,int population)
{
	String  today = text_manipulate.get_current_date_proc ();

	BasicDBObject query = new BasicDBObject("key", key_in);
	FindIterable iterable = coll.find(query);

	iterable.forEach(new Block <Document> ()
		{
		@Override
		public void apply(final Document doc) {
		final String name_aa = doc.get ("name").toString ();

		Document doc_aa = new Document("key",key_in)
		.append ("name",name_aa)
		.append ("population",population)
		.append ("date_mod",today);

		coll.replaceOne (new Document("key", key_in),doc_aa);
			}
		});
}

// --------------------------------------------------------------
public static void main (String[] args)
{
	System.out.println ("*** 開始 ***");

	String	key_in = args[0];
	int	population = Integer.parseInt (args[1]);
	System.out.print ("\tkey_in = " + key_in);
	System.out.println ("\tpopulation = " + population);

	try
	{
	MongoClient client = new MongoClient ("localhost",27017);

	MongoDatabase db = client.getDatabase ("city_db");

	String col_name = "saitama";

	MongoCollection coll = db.getCollection (col_name);
	System.out.println ("\tcoll.count ()  = " + coll.count ());

	mongo_update_proc (coll,key_in,population);

	client.close ();
	}
	catch (Exception ee)
	{
	ee.printStackTrace ();
	}

	System.out.println ("*** 終了 ***");
}

// --------------------------------------------------------------
}
// --------------------------------------------------------------

File: mongo/scala/update/mongo_update.scala
// --------------------------------------------------------------
//	update/mongo_update.scala
//
//					Sep/02/2013
// --------------------------------------------------------------

import com.mongodb.Mongo
import com.mongodb.DB
import com.mongodb.DBCollection
import com.mongodb.BasicDBObject
import com.mongodb.DBObject
import com.mongodb.DBCursor

import  scala.util.parsing.json.JSON

// --------------------------------------------------------------
object mongo_update
{

// --------------------------------------------------------------
def main (args: Array[String])
{
	println ("*** 開始 ***")

	val	key_in = args(0)
	val	population_in = args(1).toInt

	print ("\tkey_in = " + key_in)
	println ("\tpopulation_in = " + population_in)

	val mm = new Mongo ("localhost" ,27017)

	val db = mm.getDB ("city_db")

	val col_name = "saitama"

	val coll = db.getCollection (col_name)

	mongo_manipulate.mongo_update_proc (coll,key_in,population_in)

	mongo_manipulate.mongo_display_proc (coll)

	println ("*** 終了 ***")
}

// --------------------------------------------------------------
}
// --------------------------------------------------------------
